### PR TITLE
VITIS-10999 Improve electrical report output

### PR DESCRIPTION
**Pull request by dbenusov-xilinx**
#### Problem solved by the commit
https://jira.xilinx.com/browse/VITIS-10999

Data does not populate for certain devices due to bad condition for power display.

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
Introduced in https://github.com/Xilinx/XRT/pull/7862.

#### How problem was solved, alternative solutions (if any) and why they were rejected
Checked the status of each power output individually and output if valid.

#### Risks (if any) associated the changes in the commit
None.

#### What has been tested and how, request additional testing if necessary
Ubuntu 22.04 u55c
```
dbenusov@xsjdbenusov50:/proj/rdi/staff/dbenusov$ xbutil examine -d -r electrical
WARNING: Unexpected xocl version (2.16.0) was found. Expected 2.17.0, to match XRT tools.

-------------------------------------------------
[0000:04:00.1] : xilinx_u55c_gen3x16_xdma_base_3
-------------------------------------------------
Electrical
  Max Power              : 225 Watts
  Power                  : 21.445936 Watts
  Power Warning          : 225

  Power Rails            :   Voltage  Current
  ---------------------------------------------
  12 Volts Auxillary     :  12.256 V  0.481 A
  12 Volts PCI Express   :  12.240 V  0.985 A
  3.3 Volts PCI Express  :   3.360 V  1.040 A
  3.3 Volts Auxillary    :   0.000 V  0.000 A
  Internal FPGA Vcc      :   0.852 V  6.100 A
  Internal FPGA Vcc IO   :   0.855 V  3.600 A
  DDR Vpp Bottom         :   0.000 V  0.000 A
  DDR Vpp Top            :   0.000 V  0.000 A
  5.5 Volts System       :   5.003 V  0.000 A
  Vcc 1.2 Volts Top      :   0.000 V  0.000 A
  Vcc 1.2 Volts Bottom   :   0.000 V  0.000 A
  1.8 Volts Top          :   1.812 V  0.000 A
  0.9 Volts Vcc          :   0.898 V  0.000 A
  12 Volts SW            :   0.000 V  0.000 A
  Mgt Vtt                :   1.200 V  0.000 A
  3.3 Volts Vcc          :   3.354 V  0.000 A
  1.2 Volts HBM          :   1.203 V  0.000 A
  Vpp 2.5 Volts          :   2.468 V  0.000 A
  12 Volts Aux1          :   0.000 V  0.000 A
  Vcc 1.2 Volts i        :   0.000 V  0.000 A
  V12 in i               :   0.000 V  0.000 A
  V12 in Aux0 i          :   0.000 V  0.000 A
  V12 in Aux1 i          :   0.000 V  0.000 A
  Vcc Auxillary          :   0.000 V  0.000 A
  Vcc Auxillary Pmc      :   0.000 V  0.000 A
  Vcc Ram                :   0.000 V  0.000 A
  0.9 Volts Vcc Vcu      :   0.000 V  0.000 A
```

Windows VM Virtual device build 26016 (Phoenix)
```
Z:\XRT-MCDM\build\WDebug\xilinx\xrt>xbutil examine -d -r electrical
------------------------------------------------------------
                        EARLY ACCESS
        This release of xbutil contains early access
         experimental features which may have bugs.
------------------------------------------------------------

---------------------------------
[0000:18:00.1] : RyzenAI-Phoenix
---------------------------------
Electrical
  No electrical sensors found
```

Windows VM Virtual device build 26016 (Strix)
```
Z:\XRT-MCDM\build\WDebug\xilinx\xrt>xbutil examine -d -r electrical
------------------------------------------------------------
                        EARLY ACCESS
        This release of xbutil contains early access
         experimental features which may have bugs.
------------------------------------------------------------

-------------------------------
[0000:18:00.1] : RyzenAI-Strix
-------------------------------
Electrical
  Power                  : 123 Watts
```
#### Documentation impact (if any)
None

### CODE REVIEW

**Review by stsoe — APPROVED** [review]

File: src/runtime_src/core/common/sensor.cpp [hunk 1]
```diff
@@ -210,19 +210,19 @@ read_data_driven_electrical(const std::vector<xq::sdm_sensor_info::data_type>& c
     sensor_array.push_back({"", pt});
   }
 
-  uint64_t bd_power;
+  std::string bd_power = "N/A";
   // iterate over power data, store to ptree by converting to watts.
   for (const auto& tmp : power) {
     if (boost::iequals(tmp.label, "Total Power"))
-      bd_power = tmp.input;
+      bd_power = std::to_string(tmp.input);
```
Comment (pranjalv-xilinx) [inline src/runtime_src/core/common/sensor.cpp:217]: why is tmp.input not being converted to proper unit using "unit modifier" as being done for voltage and current?

File: src/runtime_src/core/common/sensor.cpp [hunk 2]
```diff
@@ -210,19 +210,19 @@ read_data_driven_electrical(const std::vector<xq::sdm_sensor_info::data_type>& c
...
   }
   ptree_type root;
 
   root.add_child("power_rails", sensor_array);
 
   root.put("power_consumption_watts", bd_power);
-  root.put("power_consumption_max_watts", "NA");
-  root.put("power_consumption_warning", "NA");
+  root.put("power_consumption_max_watts", "N/A");
```
Comment (pranjalv-xilinx) [inline src/runtime_src/core/common/sensor.cpp:224]: Also why are we not using max_watts and average consumptions being returned by driver?

File: src/runtime_src/core/common/sensor.cpp (hunk 1 above)
Comment (dbenusov-xilinx) [reply src/runtime_src/core/common/sensor.cpp:217]: Good point. We should do that.

File: src/runtime_src/core/common/sensor.cpp (hunk 2 above)
Comment (dbenusov-xilinx) [reply src/runtime_src/core/common/sensor.cpp:224]: Another good point. I will rig something up to pull out the data.

Comment (dbenusov-xilinx) [reply src/runtime_src/core/common/sensor.cpp:224]: We do not have anything for average so I will leave that one out for now (for a later PR). Max watts I will populate.

**Review by chvamshi-xilinx — APPROVED** [review]

**Review by pranjalv-xilinx — APPROVED** [review]

**Pull request merged**